tests/Mremi/UrlShortener/Tests/Provider/Baidu/BaiduProviderTest.php

```diff
@@ -0,0 +1,437 @@
+     * @expectedExceptionMessage Baidu returned status code "-1: long url is not valid
+     */
+    public function testExpandThrowsExceptionIfApiResponseIsError()
+    {
+        $this->mockClient($this->getMockResponseWithError(), 'post');
+
+        $this->provider->expand($this->getBaseMockLink());
+    }
+
+    /**
+     * Tests the expand method throws exception if Baidu returns a response with no longUrl.
+     *
+     * @expectedException        \Mremi\UrlShortener\Exception\InvalidApiResponseException
+     * @expectedExceptionMessage Baidu returned status code "-2: short url dose not exist
+     */
+    public function testExpandThrowsExceptionIfApiResponseHasNoLongUrl()
+    {
+        $this->mockClient($this->getMockResponseWithNoLongUrl(), 'post');
+
+        $this->provider->expand($this->getBaseMockLink());
+    }
+
+    /**
+     * Tests the expand method throws exception if Baidu returns a response with no status.
+     *
+     * @expectedException        \Mremi\UrlShortener\Exception\InvalidApiResponseException
+     * @expectedExceptionMessage Property "status" does not exist within Baidu response.
+     */
+    public function testExpandThrowsExceptionIfApiResponseHasNoStatus()
+    {
+        $response = $this->getBaseMockResponse();
+
+        $apiRawResponse = <<<'JSON'
+{
+"tinyurl": "http://dwz.cn/OErDnjcx",
+"longurl": "http://www.google.com/",
+"err_msg": ""
+}
+JSON;
+
+        $stream = $this->getBaseMockStream();
+        $stream
+            ->expects($this->once())
+            ->method('getContents')
+            ->will($this->returnValue($apiRawResponse));
+
+        $response
+            ->expects($this->once())
+            ->method('getBody')
+            ->will($this->returnValue($stream));
+
+        $this->mockClient($response, 'post');
+
+        $this->provider->expand($this->getBaseMockLink());
+    }
+
+    /**
+     * Tests the expand method with a valid Baidu's response.
+     */
+    public function testExpandWithValidApiResponse()
+    {
+        $response = $this->getBaseMockResponse();
+
+        $apiRawResponse = <<<'JSON'
+{
+"status": 0,
+"tinyurl": "http://dwz.cn/OErDnjcx",
+"longurl": "http://www.google.com/",
+"err_msg": ""
+}
+JSON;
+
+        $stream = $this->getBaseMockStream();
+        $stream
+            ->expects($this->once())
+            ->method('getContents')
+            ->will($this->returnValue($apiRawResponse));
+
+        $response
+            ->expects($this->once())
+            ->method('getBody')
+            ->will($this->returnValue($stream));
+
+        $link = $this->getMockShortLink();
+        $link
+            ->expects($this->once())
+            ->method('setLongUrl')
+            ->with($this->equalTo('http://www.google.com/'));
+
+        $this->mockClient($response, 'post');
+
+        $this->provider->expand($link);
+    }
+
+    /**
+     * Initializes the provider.
+     */
+    protected function setUp()
```

Comment:
Can you move it at the top of the class please (same for `tearDown`)? By convention, these methods are declared first to increase readability.